cmd/kubeadm/app/util/runtime/runtime_windows.go

```diff
@@ -20,17 +20,30 @@ limitations under the License.
 package util
 
 import (
+	"fmt"
+	"net/url"
+	"strings"
+
 	winio "github.com/Microsoft/go-winio"
+
+	"k8s.io/klog/v2"
 )
 
 const (
-	dockerSocket     = "//./pipe/docker_engine"         // The Docker socket is not CRI compatible
-	containerdSocket = "//./pipe/containerd-containerd" // Proposed containerd named pipe for Windows
+	dockerSocket     = "npipe:////./pipe/docker_engine"         // The Docker socket is not CRI compatible
+	containerdSocket = "npipe:////./pipe/containerd-containerd" // Proposed containerd named pipe for Windows
 )
 
 // isExistingSocket checks if path exists and is domain socket
 func isExistingSocket(path string) bool {
-	_, err := winio.DialPipe(path, nil)
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		// should not happen, since we are trying to access known / hardcoded sockets
+		return false
+	}
+
+	// the dial path must be without "npipe://"
+	_, err = winio.DialPipe(u.Path, nil)
```

Comment:
@pacoxu the kubelet original code that you had in your PR managed `tcp://` as well, which and i'm not sure it's actually supported in the kubelet.

if we need to support it in kubeadm we can add it eventually, but for the time being i'm only leaving `npipe://`
otherwise we have to use two separate check methods here - `DialPipe` for named pipes and `net.Dial` (?) for Windows TCP sockets. we also would have to have multiple supported URL schemes per OS.

i think kubeadm Windows users are currently only using container runtimes exposed on named pipes.

Reply:
Agree. I am not sure if TCP is supported in kubelet.
LGTM here.